### PR TITLE
Check for user 'jboss'

### LINKED ISSUE #337: Check for 'jboss' user in EAP scan
Part of #335 .

### PR DESCRIPTION
**Pull request by noahl**
Use "id -u jboss" to check if a user jboss exists on the system. This
will be one of our indicators that a system might have jboss on it.

Closes #337 .

### CODE REVIEW

File: library/spit_results.py [hunk 1]
```diff
@@ -288,6 +288,46 @@ def classify(key, fact_names, classifications):
...
+    # We use the 'id' command to check for jboss because it's been in
+    # GNU coreutils since 1992, so it should be present on every
+    # system we encounter.
+
+    if 'jboss.eap.jboss-user' not in fact_names:
+        return {}
+
+    if 'jboss_eap_id_jboss' not in host_vars:
+        return {JBOSS_EAP_JBOSS_USER:
+                'Error: "id -u jboss" not in Ansible playbook'}
```
Comment (chambridge) [inline library/spit_results.py]: Not sure this error makes sense. Not really related to a playbook, more that the value wasn't captured on the remote system.

**Review by chambridge — COMMENTED** [review]
Minor comment on error message and you need to add documentation for the new fact.

File: library/spit_results.py (hunk 1 above)
Comment (noahl) [reply library/spit_results.py]: I think if the value wasn't captured, it should appear in the output but with special error output.

However, I will change the error.

**Comment by noahl** [comment]
@chambridge thank you for catching the doc issue! I've pushed a new commit; do you think this is good?

**Review by chambridge — APPROVED** [review]
LGTM!

**Pull request merged**